File: AspNetMvcUnleashed/0672329980_source/Chapter05/VB/EFMvcApplication/EFMvcApplication.Tests/Controllers/HomeControllerTestFake.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports GenericRepository
Imports System.Web.Mvc
Imports System.Collections

<TestClass()> _
Public Class HomeControllerTestFake
    
Private _fakeRepository As IGenericRepository

    <TestInitialize()> _
    Public Sub Initialize()
        _fakeRepository = New FakeGenericRepository()
    End Sub

    <TestMethod()> _
    Public Sub CreateThenList()
        ' Arrange
        Dim controller = New HomeController(_fakeRepository)
        Dim productToCreate = Product.CreateProduct(-1, "Test", "Test", 3.44D)

        ' Act
        controller.Create(productToCreate)
        Dim results = CType(controller.Index(), ViewResult)

        ' Assert
        Dim products = CType(results.ViewData.Model, ICollection)
        CollectionAssert.Contains(products, productToCreate)
    End Sub

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter05/CS/EFMvcApplication/EFMvcApplication.Tests/Controllers/HomeControllerTestFake.cs
using System.Collections;
using System.Web.Mvc;
using EFMvcApplication.Controllers;
using EFMvcApplication.Models;
using GenericRepository;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace EFMvcApplication.Tests.Controllers
{
    [TestClass]
    public class HomeControllerTestFake
    {
        IGenericRepository _fakeRepository; 

        [TestInitialize]
        public void Initialize()
        {
            _fakeRepository = new FakeGenericRepository();
        }

        [TestMethod]
        public void CreateThenList()
        {
            // Arrange
            var controller = new HomeController(_fakeRepository);
            var productToCreate = Product.CreateProduct(-1, "Test", "Test", 3.44m); 

            // Act
            controller.Create(productToCreate);
            var results = (ViewResult)controller.Index();

            // Assert
            var products = (ICollection)results.ViewData.Model;
            CollectionAssert.Contains(products, productToCreate);
        }
    }
}
